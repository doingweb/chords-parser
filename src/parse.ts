import parser from './parser';
import Song from './lib/song';

export default function parse(text: string): Song {
  return new Song(parser.parse(text), text);
}
