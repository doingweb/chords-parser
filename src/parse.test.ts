import { readdirSync, readFileSync } from 'fs';
import * as path from 'path';
import parse from './parse';

describe('basic examples', () => {
  test.each([
    ['just chords', 'C Cmaj7 Am F G'],
    ['just lyrics', 'test test test'],
    ['just a heading', '[Verse]'],
    ['empty', ''],
  ])('%s', (_, songText) => {
    expect(parse(songText)).toMatchSnapshot();
  });
});

describe('fixtures', () => {
  const fixturesPath = path.join(__dirname, '__fixtures__');
  const filenames = readdirSync(fixturesPath);

  test.each(filenames)('%s', (filename) => {
    const songText = readFileSync(path.join(fixturesPath, filename), 'utf8');
    expect(parse(songText)).toMatchSnapshot();
  });
});
