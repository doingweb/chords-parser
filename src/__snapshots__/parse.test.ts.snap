// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic examples empty 1`] = `
Song {
  "lines": Array [
    EmptyLine {
      "type": "empty",
    },
  ],
  "originalText": "",
}
`;

exports[`basic examples just a heading 1`] = `
Song {
  "lines": Array [
    SectionHeading {
      "content": "Verse",
      "type": "heading",
    },
  ],
  "originalText": "[Verse]",
}
`;

exports[`basic examples just chords 1`] = `
Song {
  "lines": Array [
    ChordLine {
      "chords": Array [
        Chord {
          "content": "C",
          "position": 0,
        },
        Chord {
          "content": "Cmaj7",
          "position": 2,
        },
        Chord {
          "content": "Am",
          "position": 8,
        },
        Chord {
          "content": "F",
          "position": 11,
        },
        Chord {
          "content": "G",
          "position": 13,
        },
      ],
      "type": "chords",
    },
  ],
  "originalText": "C Cmaj7 Am F G",
}
`;

exports[`basic examples just lyrics 1`] = `
Song {
  "lines": Array [
    Lyric {
      "content": "test test test",
      "offset": 0,
      "type": "lyric",
    },
  ],
  "originalText": "test test test",
}
`;

exports[`fixtures readme-example.txt 1`] = `
Song {
  "lines": Array [
    SectionHeading {
      "content": "Verse",
      "type": "heading",
    },
    ChordLine {
      "chords": Array [
        Chord {
          "content": "G",
          "position": 0,
        },
        Chord {
          "content": "C",
          "position": 8,
        },
        Chord {
          "content": "D",
          "position": 18,
        },
        Chord {
          "content": "G",
          "position": 31,
        },
      ],
      "type": "chords",
    },
    Lyric {
      "content": "This is where the lyrics would go",
      "offset": 0,
      "type": "lyric",
    },
    EmptyLine {
      "type": "empty",
    },
  ],
  "originalText": "[Verse]
G       C         D            G
This is where the lyrics would go
",
}
`;
